contracts/utils/contracts/ScaleCodec.sol

```diff
@@ -54,6 +56,28 @@ library ScaleCodec {
         }
     }
 
+
+    function encodeUintCompact(uint256 v) internal pure returns (bytes memory) {
+        if ( v >= 0 && v <= 63 ) {
+            return abi.encodePacked(uint8(v << 2));
+        } else if ( v > 63 && v <= (2 ** 14) - 1 ) {
+            return abi.encodePacked(reverse16(uint16(((v << 2) + 1))));
+        } else if ( v > (2 ** 14) - 1 && v <= (2 ** 30) - 1 ) {
+            return abi.encodePacked(reverse32(uint32(((v << 2) + 2))));
+        } else if ( v > 2 ** 30 - 1 ) {
+            bytes memory valueBytes = 
+                Bytes.removeEndingZero(abi.encodePacked(reverse256(v)));
+
+            uint length = valueBytes.length;
+            bytes memory prefix = 
+                Bytes.removeLeadingZero(abi.encodePacked(((length - 4) << 2) + 3));
```

Comment:
if there are leading zeros, you need to remove these zeros.

but the big mode is bigger than 256, so the code here is not perfect.

Reply:
the biggest compact is 2 ** 536 - 1. It's ok now.

Reply:
1. 0 <= v < 2**256
2. Even in big-inter mode, prefix is also `bytes1`.  [ 11 _ _ _ _ ]
    cause this, it can only represent 2**536 - 1.
    (2 ** 6 + 4 - 1) * 8 == 536.
So. this is no need to use `uint256` present the length, `uint8` is enough.

Reply:
I get it. the first bit of uint8 is not zero.

Reply:
But after changed, the test failed

Reply:
Sorry, I forget the operator order and change the suggetion.